communication/mavlinkparameterserver.h

```diff
@@ -10,9 +15,13 @@ class MavlinkParameterServer : public ParameterServer
 {
     Q_OBJECT
 public:
-    explicit MavlinkParameterServer(std::shared_ptr<mavsdk::ServerComponent> serverComponent);
+    static void createSingleton(std::shared_ptr<mavsdk::ServerComponent> serverComponent);
```

Comment:
Unsure whether to call it "createSingleton" or better "init"/"initialize": the users of this class mainly need to be concerned about that it only can be created/initialized once, but might be a good idea to make it explicit that it indeed is a singleton... you decide. 😉